pkg/scalers/azure_eventhub.go

```diff
@@ -0,0 +1,131 @@
+package scalers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	eventhub "github.com/Azure/azure-event-hubs-go"
+	storageLeaser "github.com/Azure/azure-event-hubs-go/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/Azure/go-autorest/autorest/azure"
+	log "github.com/Sirupsen/logrus"
+)
+
+// GetStorageCredentials returns azure env and storage credentials
+func GetStorageCredentials(storageConnection string) (azure.Environment, *azblob.SharedKeyCredential, error) {
+	storageAccountName, storageAccountKey, err := parseAzureStorageConnectionString(storageConnection)
+	if err != nil {
+		return azure.Environment{}, &azblob.SharedKeyCredential{}, fmt.Errorf("unable to parse connection string: %s\n", storageConnection)
+	}
+
+	azureEnv, err := azure.EnvironmentFromName("AzurePublicCloud")
+	if err != nil {
+		return azureEnv, nil, fmt.Errorf("could not get azure.Environment struct: %s", err)
+	}
+
+	cred, err := azblob.NewSharedKeyCredential(storageAccountName, storageAccountKey)
+	if err != nil {
+		return azureEnv, nil, fmt.Errorf("could not prepare a blob storage credential: %s", err)
+	}
+
+	return azureEnv, cred, nil
+}
+
+// GetLeaserCheckpointer gets the leaser/checkpointer using storage credentials
+func GetLeaserCheckpointer(storageConnection string, storageContainerName string) (*storageLeaser.LeaserCheckpointer, error) {
+	storageAccountName, _, err := ParseAzureStorageConnectionString(storageConnection)
+	if err != nil {
+		return &storageLeaser.LeaserCheckpointer{}, fmt.Errorf("unable to parse storage connection string: %s", err)
+	}
+
+	env, cred, err := GetStorageCredentials(storageConnection)
+	if err != nil {
+		return &storageLeaser.LeaserCheckpointer{}, fmt.Errorf("unable to get storage credentials: %s", err)
+	}
+
+	leaserCheckpointer, err := storageLeaser.NewStorageLeaserCheckpointer(
+		cred,
+		storageAccountName,
+		storageContainerName,
+		env)
+	if err != nil {
+		return nil, fmt.Errorf("could not prepare a storage leaserCheckpointer: %s", err)
+	}
+
+	return leaserCheckpointer, nil
+}
+
+// GetEventHubClient returns eventhub client
+func GetEventHubClient(connectionString string) (*eventhub.Hub, error) {
+	hub, err := eventhub.NewHubFromConnectionString(connectionString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create hub client: %s", err)
+	}
+
+	return hub, nil
+}
+
+// GetLeaseFromBlobStorage accesses Blob storage and gets lease information of a partition
+func GetLeaseFromBlobStorage(ctx context.Context, partitionID string, storageConnection string, storageContainerName string) (Lease, error) {
```

Comment:
Isn't this more last checkpoint from Bob storage?

Reply:
The Checkpoint is included in the Lease. In the library in golang, it's called a LeaserCheckpointer so both are included. Should I change the name?